### PR TITLE
feat(cross-filters): add option to clear set cross filters

### PR DESCRIPTION
**Pull request by villebro**
### SUMMARY
This adds the option to clear emitted cross filters by adding an `onClick` to the cross filter indicator. Also fixes 

### AFTER
image

### BEFORE
image

### TESTING INSTRUCTIONS
1. Enable the `DASHBOARD_CROSS_FILTERS` feature flag
2. create a chart that supports cross filters and check the "emit cross filters" control
3. add the chart to a dashboard
4. emit a cross filter
5. Hover on the cross filter indicator, see the updated tooltip and click on the indicator to clear emitted filters

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by michael-s-molina** [comment]
/testenv up FEATURE_DASHBOARD_CROSS_FILTERS=true FEATURE_DASHBOARD_NATIVE_FILTERS=true

**Comment by amitmiran137** [comment]
/testenv up FEATURE_DASHBOARD_CROSS_FILTERS=true FEATURE_DASHBOARD_NATIVE_FILTERS=true

**Comment by villebro** [comment]
/testenv up FEATURE_DASHBOARD_CROSS_FILTERS=true FEATURE_DASHBOARD_NATIVE_FILTERS=true

**Comment by michael-s-molina** [comment]
@villebro Some suggestions:
- Vertical align the emit icon and the filter icon
- Make the emit icon and the filter icon the same size (emit is bigger)
- I think "Click to clear emitted filter" does not need to be bold because it's secondary information. After the first interaction, the user won't even read that anymore.

File: superset-frontend/src/dashboard/components/FiltersBadge/Styles.tsx [hunk 1]
```diff
@@ -153,3 +153,12 @@ export const FilterValue = styled.div`
...
 `;
+
+export const FilterIndicatorText = styled.div`
+  padding-top: ${({ theme }) => theme.gridUnit * 3}px;
+  font-weight: ${({ theme }) => theme.typography.weights.bold};
+  max-width: 100%;
+  flex-grow: 1;
+  overflow: auto;
+  color: ${({ theme }) => theme.colors.grayscale.light5};
+`;
```
Comment (michael-s-molina) [inline superset-frontend/src/dashboard/components/FiltersBadge/Styles.tsx:164]:
```suggestion

export const FilterIndicatorText = styled.div`
  ${({ theme }) => `
    padding-top: ${theme.gridUnit * 3}px;
    font-weight: ${theme.typography.weights.bold};
    max-width: 100%;
    flex-grow: 1;
    overflow: auto;
    color: ${theme.colors.grayscale.light5};
  `}
`;
```

**Comment by villebro** [comment]
> @villebro Some suggestions:
> 
> * Vertical align the emit icon and the filter icon
> * Make the emit icon and the filter icon the same size (emit is bigger)

Good catch, I hadn't noticed that.

> * I think "Click to clear emitted filter" does not need to be bold because it's secondary information. After the first interaction, the user won't even read that anymore.

I agree, will make the change

**Comment by villebro** [comment]
/testenv up FEATURE_DASHBOARD_CROSS_FILTERS=true FEATURE_DASHBOARD_NATIVE_FILTERS=true

**Review by michael-s-molina — APPROVED** [review]
LGTM

**Pull request merged**